### PR TITLE
Fix Storm local file seeking (also fixes `SBmpLoadImage`)

### PR DESCRIPTION
**Pull request by glebm**
SBmpLoadImage tried to get the size of the rest of the file incorrectly
by subtracting the current file position from the total size.

This was incorrect because file positions do not always begin at 0.

The only semi-portable way to get a size from file positions is to
subtract two file positions.

### CODE REVIEW

File: SourceX/storm/storm.cpp [hunk 1]
```diff
@@ -228,7 +246,7 @@ BOOL SBmpLoadImage(const char *pszFileName, SDL_Color *pPalette, BYTE *pBuffer,
 	}
 
 	if (pPalette && pcxhdr.BitsPerPixel == 8) {
-		SFileSetFilePointer(hFile, -768, 0, 1);
+		SFileSetFilePointer(hFile, -768, NULL, DVL_FILE_CURRENT);
```
Comment (AJenbo) [inline SourceX/storm/storm.cpp:249]: is DVL_FILE_CURRENT the proper def to use when talking to storm?

Comment (glebm) [reply SourceX/storm/storm.cpp:249]: We don't have any other defs for this as I understand

Comment (glebm) [reply SourceX/storm/storm.cpp:249]: I thought you defined these specifically to talk to Storm?

**Comment by AJenbo** [comment]
Still not working :)

image

**Comment by glebm** [comment]
Fixed and sent a PR to fix StormLib upstream as well https://github.com/ladislav-zezula/StormLib/pull/204

**Pull request merged**